render/detailed/links.go

```diff
@@ -0,0 +1,230 @@
+		},
+		// Pod naming:
+		// https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label
+		"__k8s_controllers": {
+			docker.MemoryUsage:   parsedTemplate(`sum(container_memory_usage_bytes{pod_name=~"^{{.Label}}-[^-]+-[^-]+$"})`),
+			docker.CPUTotalUsage: parsedTemplate(`sum(rate(container_cpu_usage_seconds_total{pod_name=~"^{{.Label}}-[^-]+-[^-]+$"}[1m]))`),
+		},
+		report.DaemonSet: {
+			docker.MemoryUsage:   parsedTemplate(`sum(container_memory_usage_bytes{pod_name=~"^{{.Label}}-[^-]+$"})`),
+			docker.CPUTotalUsage: parsedTemplate(`sum(rate(container_cpu_usage_seconds_total{pod_name=~"^{{.Label}}-[^-]+$"}[1m]))`),
+		},
+		report.Service: {
+			// These recording rules must be defined in the prometheus config.
+			// NB: Pods need to be labeled and selected by their respective Service name, meaning:
+			// - The Service's `spec.selector` needs to select on `name`
+			// - The Service's `metadata.name` needs to be the same value as `spec.selector.name`
+			docker.MemoryUsage:   parsedTemplate(`namespace_label_name:container_memory_usage_bytes:sum{label_name="{{.Label}}"}`),
+			docker.CPUTotalUsage: parsedTemplate(`namespace_label_name:container_cpu_usage_seconds_total:sum_rate{label_name="{{.Label}}"}`),
+		},
+	}
+	k8sControllers = map[string]struct{}{
+		report.Deployment: {},
+		"stateful_set":    {},
+		"cron_job":        {},
+	}
+)
+
+// SetMetricsGraphURL sets the URL we deduce our eventual metric link from.
+// Supports placeholders such as `:orgID` and `:query`. An empty url disables
+// this feature. If the `:query` part is missing, a JSON version will be
+// appended, see `queryParamsAsJSON()` for more info.
+func SetMetricsGraphURL(url string) {
+	metricsGraphURL = url
+}
+
+// RenderMetricURLs sets respective URLs for metrics in a node summary. Missing metrics
+// where we have a query for will be appended as an empty metric (no values or samples).
+func RenderMetricURLs(summary NodeSummary, n report.Node) NodeSummary {
+	if metricsGraphURL == "" {
+		return summary
+	}
+
+	queries := getTopologyQueries(n.Topology)
+	if len(queries) == 0 {
+		return summary
+	}
+
+	var maxprio float64
+	var bs bytes.Buffer
+	var ms []report.MetricRow
+	found := make(map[string]struct{})
+
+	// Set URL on existing metrics
+	for _, metric := range summary.Metrics {
+		if metric.Priority > maxprio {
+			maxprio = metric.Priority
+		}
+		tpl := queries[metric.ID]
+		if tpl == nil {
+			continue
+		}
+
+		bs.Reset()
+		if err := tpl.Execute(&bs, summary); err != nil {
+			continue
+		}
+
+		ms = append(ms, metric)
+		ms[len(ms)-1].URL = buildURL(bs.String())
+		found[metric.ID] = struct{}{}
+	}
+
+	// Append empty metrics for unattached queries
+	for _, metadata := range shownQueries {
+		if _, ok := found[metadata.ID]; ok {
+			continue
+		}
+
+		tpl := queries[metadata.ID]
+		if tpl == nil {
+			continue
+		}
+
+		bs.Reset()
+		if err := tpl.Execute(&bs, summary); err != nil {
```

Comment:
Good point, I started off thinking there might be other parts in there that will be used but it now looks like we only care about the `label` and nothing else. Will take a look